### PR TITLE
DAOS-5298 cleanup and fix for EC aggregation

### PR DESCRIPTION
**Pull request by gnailzenh**
- Add inline functions to ease access to structure members
- remove debug code from do_obj_ioc_begin
- reduce code indentation of a few functions

Fixes:
- bug fix for the loop in agg_get_obj_handle

Signed-off-by: Liang Zhen <liang.zhen@intel.com>

### CODE REVIEW

File: src/object/srv_ec_aggregate.c [hunk 1]
```diff
@@ -329,6 +361,20 @@ agg_alloc_buf(d_sg_list_t *sgl, size_t ent_buf_len, unsigned int iov_entry,
 	return rc;
 }
 
+/* Recovers allocated memory for sgl used in the aggregation process.
+ */
+static void
+agg_sgl_fini(d_sg_list_t *sgl)
```
Comment (liuxuezhao) [inline src/object/srv_ec_aggregate.c:367]: can reuse d_sgl_fini(sgl, true);

File: src/object/srv_ec_aggregate.c [hunk 2]
```diff
@@ -1646,10 +1665,10 @@ agg_process_stripe(struct ec_agg_entry *entry)
 	iter_param.ip_hdl		= DAOS_HDL_INVAL;
 	iter_param.ip_ih		= entry->ae_thdl;
 	iter_param.ip_flags		= VOS_IT_RECX_VISIBLE;
+	iter_param.ip_recx.rx_nr	= ec_age2cs(entry);
 	iter_param.ip_recx.rx_idx	= PARITY_INDICATOR |
 					  (entry->ae_cur_stripe.as_stripenum *
-						entry->ae_oca->u.ec.e_len);
-	iter_param.ip_recx.rx_nr	= entry->ae_oca->u.ec.e_len;
+					   iter_param.ip_recx.rx_idx);
```
Comment (liuxuezhao) [inline src/object/srv_ec_aggregate.c:1671]: L1671 seems typo/wrong? (should be iter_param.ip_recx.rx_nr, or ec_age2cs(entry))

Comment (liuxuezhao) [reply src/object/srv_ec_aggregate.c:1671]: BTW, ec_age2cs(entry) looks better, although they are same.

**Review by jgmoore-or — DISMISSED** [review]

File: src/object/srv_ec_aggregate.c (hunk 2 above)
Comment (gnailzenh) [reply src/object/srv_ec_aggregate.c:1671]: indeed, thanks

**Comment by gnailzenh** [comment]
hmm, need to refresh again because of conflict

**Review by jgmoore-or — DISMISSED** [review]

**Review by jgmoore-or — APPROVED** [review]

**Review by liuxuezhao — APPROVED** [review]

**Comment by gnailzenh** [comment]
HDF5 VOL issue has been fixed

**Pull request merged**